uaclient/update_info.py

```diff
@@ -0,0 +1,88 @@
+from typing import Any, Dict
+
+from uaclient import util
+from uaclient.config import UAConfig
+
+# We are doing this because we are sure that python3-apt comes with the distro,
+# but it cannot be installed in a virtual environment to be properly tested.
+# All our unit tests are mocking the Cache object, so we are not losing
+# anything here. If by any chance there is no python3-apt in the future, the
+# integration tests should be able to detect it through failure, because we
+# don't expect an empty apt cache there.
+try:
+    from apt import Cache  # type: ignore
+except ImportError:
+
+    def Cache():
+        return []
+
+
+series = util.get_platform_info()["series"]
+
+
+def archive_to_service(archive):
+    if archive in (
+        "{}-infra-updates".format(series),
+        "{}-infra-security".format(series),
+    ):
+        return "esm-infra"
+    if archive in (
+        "{}-apps-updates".format(series),
+        "{}-apps-security".format(series),
+    ):
+        return "esm-apps"
+    return archive
+
+
+def get_service_status(status, service_name):
+    for service in status["services"]:
+        if service["name"] == service_name:
+            return (service["entitled"], service["status"])
+
+
+def update_info(cfg: UAConfig) -> Dict[str, Any]:
+    output = {
+        "summary": {
+            "attached": False,
+            "enabled_services": [],
+            "entitled_services": [],
+        },
+        "upgradable_packages": [],
+    }  # type: Dict[str, Any]
+    cache = Cache()
+    status = cfg.status()
+
+    is_attached = status.get("attached")
+    output["summary"]["attached"] = is_attached
+
+    if is_attached:
+        for service in ("esm-infra", "esm-apps"):
+            entitled, enabled = get_service_status(status, service)
+            if entitled == "yes":
+                output["summary"]["entitled_services"].append(service)
+            if enabled == "enabled":
+                output["summary"]["enabled_services"].append(service)
+
+    upgradable_packages = [
```

Comment:
Maybe we should create auxiliary functions for the different parts of the code to easily test the individual pieces here

Reply:
I am sure we will need to do it in future iterations. As of now, I'd like to keep it as is, because it represents well which kind of data is being defined for the output and there is a single flow. Feel free to point out to pieces you think that can be isolated.